merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/services/TypescriptCodeGenerationServiceTest.java

```diff
@@ -1,17 +1,82 @@
 package gov.nasa.jpl.aerie.merlin.server.services;
 
+import gov.nasa.jpl.aerie.merlin.server.mocks.StubMissionModelService;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class TypescriptCodeGenerationServiceTest {
 
   @Test
-  void testCodeGen() {
+  void testCodeGen() throws MissionModelService.NoSuchMissionModelException {
+    final var codeGenService = new TypescriptCodeGenerationService(new StubMissionModelService());
+
     assertEquals(
         """
             /** Start Codegen */
+            import * as AST from './constraints-ast.js';
+            import { Discrete, Real, Windows } from './constraints-edsl-fluent-api.js';
+            export type ActivityTypeName =
+              | "activity"
+              ;
+            export type ResourceName = "mode" | "state of charge";
+            export type DiscreteResourceSchema<R extends ResourceName> =
+              R extends "mode" ? ( | "Option1" | "Option2") :
+              R extends "state of charge" ? number :
+              void;
```

Comment:
I don't think void should be used here, but rather undefined. https://stackoverflow.com/a/58885486

Reply:
done